File: src/main/java/io/github/tanguygab/conditionalactions/actions/messages/TitleAction.java
package io.github.tanguygab.conditionalactions.actions.messages;

import io.github.tanguygab.conditionalactions.actions.Action;
import org.bukkit.OfflinePlayer;
import org.bukkit.entity.Player;

public class TitleAction extends Action {

    public TitleAction() {
        super("(?i)title:( )?");
    }

    @Override
    public String getSuggestion() {
        return getSuggestionWithArgs("title: <title>","<subtitle>","<fadein>","<stay>","<fadeout>");
    }

    @Override
    public void execute(OfflinePlayer player, String match) {
        if (!(player instanceof Player p)) return;
        match = parsePlaceholders(p,match);
        String[] matches = split(match);

        String title = matches[0];
        String subtitle = "";
        int fadein = 5;
        int stay = 5;
        int fadeout = 5;
        if (matches.length > 1) subtitle = matches[1];
        if (matches.length > 2) fadein = parseInt(matches[2]);
        if (matches.length > 3) stay = parseInt(matches[3]);
        if (matches.length > 4) fadeout = parseInt(matches[4]);

        p.sendTitle(title,subtitle,fadein,stay,fadeout);
    }

    private int parseInt(String str) {
        return parseInt(str,5);
    }
}

File: src/main/kotlin/io/github/tanguygab/conditionalactions/actions/types/messages/TitleAction.kt
package io.github.tanguygab.conditionalactions.actions.types.messages

import io.github.tanguygab.conditionalactions.actions.Action
import net.kyori.adventure.text.Component
import net.kyori.adventure.title.Title
import org.bukkit.OfflinePlayer
import org.bukkit.entity.Player

class TitleAction : Action("^(?i)title:( )?") {
    override fun getSuggestion() = getSuggestionWithArgs("title: <title>", "<subtitle>", "<fadein>", "<stay>", "<fadeout>")

    override fun execute(player: OfflinePlayer?, match: String) {
        if (player !is Player) return
        val args = split(parsePlaceholders(player, match))

        val title = mm.deserialize(args[0])
        val subtitle = if (args.size > 1) mm.deserialize(args[1]) else Component.empty()

        if (args.size < 3) {
            player.showTitle(Title.title(title, subtitle))
            return
        }

        val fadein = args[2].toIntOrNull() ?: 10
        val stay = args[3].toIntOrNull() ?: 70
        val fadeout = args[4].toIntOrNull() ?: 20
        player.showTitle(Title.title(title, subtitle, fadein, stay, fadeout))
    }
}
